docs/spec_draft.md

```diff
@@ -26,6 +26,9 @@ Following are the supported element types in StableHLO:
     (represents a pair of `f64`). Exact representation of complex types
     (e.g. whether the real part or the imaginary part comes first in memory)
     is implementation-defined.
+  * **Token Types** Values of this type are used for ordering side-effecting
+    operations using data dependencies which are robust against compiler
+    optimizations.
```

Comment:
Proposed rewording:

```
  * **Token type** Values of this type are used for imposing order on execution
    of side-effecting operations using data dependencies.
```

Feedback:
  1) How about "Token type", i.e. singular form, since there's only one token type? Same for "Boolean type" above.
  2) The conventional capitalization is "Token type", i.e. capital "Token" and lowercase "type".
  3) Arguably, the operations are already ordered when they are put together in
  functions. What is ordered is execution of these operations.
  4) Let's drop "which are robust against compiler optimizations". Not all transformations are optimizations. And use cases of tokens aren't limited to just transformations (e.g. there are also analyses).